src/templates/parseJavaTemplates.ts

```diff
@@ -0,0 +1,192 @@
+                for (const validator of rawSetting.validators) {
+                    if (!value || value.match(validator.expression) === null) {
+                        return getVariableValue(resources, variables, validator.errorText);
+                    }
+                }
+            }
+
+            return undefined;
+        }
+    };
+}
+
+function getVariableValue(resources: IResources, variables: IVariables, data: string): string {
+    if (!isString(data)) {
+        // This evaluates to a non-string value in rare cases, in which case we just return the value as-is
+        return data;
+    }
+
+    const matches: RegExpMatchArray | null = data.match(/\[variables\(\'(.*)\'\)\]/);
+    data = matches !== null ? variables[matches[1]] : data;
+
+    return getResourceValue(resources, <string>data);
+}
+
+function getResourceValue(resources: IResources, data: string): string {
+    const matches: RegExpMatchArray | null = data.match(/\$(.*)/);
+    return matches !== null ? resources.en[matches[1]] : data;
+}
+
+export async function parseJavaTemplates(allTemplates: IFunctionTemplate[], functionAppPath: string, telemetryProperties?: TelemetryProperties): Promise<IFunctionTemplate[]> {
+    const javaScriptTemplates: IFunctionTemplate[] = allTemplates.filter((t: IFunctionTemplate) => t.language === ProjectLanguage.JavaScript);
+    // Currently we leverage JS Script templates to get the function metadata of Java Functions.
+    // Will refactor the code here when templates HTTP API is ready.
+    // See issue here: https://github.com/Microsoft/vscode-azurefunctions/issues/84
+    const basicJavaTemplates: IFunctionTemplate[] = javaScriptTemplates.filter((t: IFunctionTemplate) => basicJavaTemplateNames.find((vt: string) => vt === removeLanguageFromId(t.id)));
+    let embeddedTemplates: object[] = [];
+    try {
+        // Try to get the templates information by calling 'mvn azure-functions:list'.
+        const commandResult: string = await mavenUtils.executeMvnCommand(telemetryProperties, undefined, functionAppPath, 'azure-functions:list');
+        const regExp: RegExp = /(?:>> templates begin <<$)([\S\s]+)(?:^\[INFO\] >> templates end <<)/gm;
+        const regExpResult: RegExpExecArray | null = regExp.exec(commandResult);
+        if (regExpResult && regExpResult.length > 1) {
+            embeddedTemplates = <object[]>JSON.parse(regExpResult[1]);
+        }
+    } catch (error) {
+        // Swallow the exception if the plugin do not support list templates information.
+        if (telemetryProperties) {
+            telemetryProperties.parseJavaTemplateErrors = parseError(error).message;
+        }
+    }
+    const templates: IFunctionTemplate[] = [];
+    const cachedResources: object | undefined = ext.context.globalState.get<object>(`FunctionTemplateResources.${ProjectRuntime.beta}`);
+    const cachedConfig: object | undefined = ext.context.globalState.get<object>(`FunctionTemplateConfig.${ProjectRuntime.beta}`);
```

Comment:
So this information isn't included in the maven list function? That means there could be a mismatch if maven has older templates but the config is newer/different. I'm fine with that risk if you are, but I would add a comment explaining that

Reply:
Emm, that's a good question. Cache the bindings.json & Resources.json into Maven plugin seems much safer. Let's me update the Maven side first.